src/main/java/com/google/api/gax/grpc/ChannelProvider.java

```diff
@@ -0,0 +1,49 @@
+package com.google.api.gax.grpc;
+
+import com.google.api.gax.core.ConnectionSettings;
+
+import io.grpc.ManagedChannel;
+
+import java.io.IOException;
+import java.util.concurrent.Executor;
+
+import javax.annotation.Nullable;
+import javax.naming.OperationNotSupportedException;
+
+/**
+ * Provides an interface to hold and build the channel that will be used. If the channel does not
+ * already exist, it will be constructed when {@link #getChannel} is called.
+ *
+ * Implementations of {@link ChannelProvider} may choose to create a new {@link ManagedChannel} for
+ * each call to {@link #getChannel}, or may return a fixed {@link ManagedChannel} instance. In cases
+ * where the same {@link ManagedChannel} instance is returned, for example by a
+ * {@link ChannelProvider} created using the
+ * {@link ServiceApiSettings#provideChannelWith(ManagedChannel, boolean)} method, and
+ * shouldAutoClose returns true, the {@link #getChannel} method will throw an
+ * {@link OperationNotSupportedException} if it is called more than once. This is to prevent the
+ * same {@link ManagedChannel} being closed prematurely when it is used by multiple client objects.
+ */
+public interface ChannelProvider {
+  /**
+   * Connection settings used to build the channel. If a channel is provided directly this will be
+   * set to null.
+   */
+  @Nullable
+  ConnectionSettings connectionSettings();
+
+  /**
+   * Indicates whether the channel should be closed by the containing API class.
+   */
+  boolean shouldAutoClose();
+
+  /**
+   * Get the channel to be used to connect to the service. The first time this is called, if the
+   * channel does not already exist, it will be created.
+   *
+   * If the {@link ChannelProvider} is configured to return a fixed {@link ManagedChannel} object
+   * and to return shouldAutoClose as true, then after the first call to {@link #getChannel},
+   * subsequent calls should throw an {@link OperationNotSupportedException}. See interface level
+   * docs for {@link ChannelProvider} for more details.
+   */
+  ManagedChannel getChannel(Executor executor) throws IOException, OperationNotSupportedException;
```

Comment:
I would prefer IllegalStateException over OperationNotSupportedException; IllegalStateException is a runtime exception, so we don't have to declare that we throw it (but still documenting it is nice).